packages/@aws-cdk/aws-apigateway/lib/model.ts

```diff
@@ -1,46 +1,151 @@
+   *
+   * The reason we need the RestApi object itself and not just the ID is because the model
+   * is being tracked by the top-level RestApi object for the purpose of calculating it's
+   * hash to determine the ID of the deployment. This allows us to automatically update
+   * the deployment when the model of the REST API changes.
+   */
+  readonly restApi: IRestApi;
 }
 
-// TODO: Implement Model, enabling management of custom models.
\ No newline at end of file
+export class Model extends Resource implements IModel {
+  /**
+   * Represents a reference to a REST API's Error model, which is available
+   * as part of the model collection by default. This can be used for mapping
+   * error JSON responses from an integration to a client, where a simple
+   * generic message field is sufficient to map and return an error payload.
+   *
+   * Definition
+   * {
+   *   "$schema" : "http://json-schema.org/draft-04/schema#",
+   *   "title" : "Error Schema",
+   *   "type" : "object",
+   *   "properties" : {
+   *     "message" : { "type" : "string" }
+   *   }
+   * }
+   */
+  public static readonly ERROR_MODEL: IModel = ({ modelName: "Error", stack: null, node: null } as unknown) as IModel;
+
+  /**
+   * Represents a reference to a REST API's Empty model, which is available
+   * as part of the model collection by default. This can be used for mapping
+   * JSON responses from an integration to what is returned to a client,
+   * where strong typing is not required. In the absence of any defined
+   * model, the Empty model will be used to return the response payload
+   * unmapped.
+   *
+   * Definition
+   * {
+   *   "$schema" : "http://json-schema.org/draft-04/schema#",
+   *   "title" : "Empty Schema",
+   *   "type" : "object"
+   * }
+   *
+   * @see https://docs.amazonaws.cn/en_us/apigateway/latest/developerguide/models-mappings.html#models-mappings-models
+   */
+  public static readonly EMPTY_MODEL: IModel = ({ modelName: "Empty", stack: null, node: null } as unknown) as IModel;
+
+  public static fromModelName(scope: Construct, id: string, modelName: string): IModel {
+    class Import extends Resource implements IModel {
+      public readonly modelName = modelName;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * A name for the model. If you don't specify a name, AWS CloudFormation
+   * generates a unique physical ID and uses that ID for the model name.
+   * For more information, @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-name.html.
+   *
+   * Important
+   * If you specify a name, you cannot perform updates that require replacement
+   * of this resource. You can perform updates that require no or some interruption.
+   * If you must replace the resource, specify a new name.
+   *
+   * @attribute
+   */
+  public readonly modelName: string;
+
+  constructor(scope: Construct, id: string, props: ModelProps) {
+    super(scope, id);
```

Comment:
Follow the new pattern to support physical names (see s3.Bucket for example).

Reply:
done